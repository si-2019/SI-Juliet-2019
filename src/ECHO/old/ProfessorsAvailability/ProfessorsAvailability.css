.professors-table{
    margin-top: 75px;
}
.professors-table>thead>tr>th:first-child{
    width:calc(450%/12);
}
.professors-table>thead>tr>th:nth-child(2){
    width:calc(450%/12);
}
.professors-table>thead>tr>th:nth-child(3){
    width:calc(200%/12);
}
.professors-table>thead>tr>th:last-child{
    width:calc(100%/12);
}
.professors-table>tbody>tr>td:last-child{
    text-align: center;
    color: #9b9b9b;
 }
.professors-table>tbody>tr>td>a{
    color: #9b9b9b;
}
.shadow-border{
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.30);
    border-radius: 4px;
}
.search-teachers{
    margin-top: 100px;
}
.search-teachers>input{
    border: none;
    height: 50px;
    width:100%;
    padding-left: 10px;
}
.pagination-table{
    position: absolute;
    top: 850px;
    left: 50%;
}