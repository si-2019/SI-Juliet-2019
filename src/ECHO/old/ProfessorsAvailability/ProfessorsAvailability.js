import React, {Component, Fragment} from 'react';
import EditIcon from '@material-ui/icons/Edit';
import Pagination from 'rc-pagination'
import localeInfo from 'rc-pagination/lib/locale/en_US';
import './ProfessorsAvailability.css';
import 'rc-pagination/assets/index.css';
import CircularProgress from '@material-ui/core/CircularProgress';

class ProfessorsAvailability extends Component {
    constructor(props){
        super(props);
        this.state={
            professorsList: [],
            page: 1,
            size: 10,
            total: 0,
            input: '',
        }
    }
    onChangeHandler(e){
        this.setState({
            input: e.target.value,
            page: 1,
        });
        this.componentDidMount();
    }
    onChange(current, pageSize){
        this.setState({
           page: current,
        }, this.componentDidMount)
    }
    static onStaffEdit(e){
        console.log(e.target.value);
    }
    componentDidMount() {
        fetch("http://localhost:31905/si2019/echo/getTeachingStaff", {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body:JSON.stringify({
                'page': this.state.page-1,
                'size': this.state.size,
                'search': this.state.input,
            })
        })
            .then(res => {return res.json()})
            .then(
                (result) => {
                    this.setState({
                        professorsList: result.content,
                        total: result.totalElements,

                    });
                },
                (error) => {
                    this.setState({
                        isLoaded: true,
                        error
                    });
                }
            )
    }

    render() {
        var professors = this.state.professorsList
            .map((prof) => {
            return(
                <Fragment key={prof.id}>
                    <tr>
                        <td>{prof.ime} {prof.prezime}</td>
                        <td>{prof.email}</td>
                        <td>{prof.titula}</td>
                        <td><a onClick={ProfessorsAvailability.onStaffEdit.bind(prof)}><EditIcon/></a></td>
                    </tr>
                </Fragment>
            )
        });
        return (
            <div className="container">
                <div className="row search-teachers shadow-border">
                    <input id="searchTeachingStaff" type="text" placeholder="Search.." onChange={this.onChangeHandler.bind(this)}/>

                </div>
                <div className="row">
                    <table id="teachingStaffTable" className="table table-bordered table-hover professors-table shadow-border">
                        <thead>
                        <tr>
                            <th>Name</th>
                            <th>Email</th>
                            <th>Title</th>
                            <th></th>
                        </tr>
                        </thead>
                        <tbody>
                            {professors}
                        </tbody>
                    </table>
                </div>
                <Pagination
                    id="pagination" className="pagination-table"
                    current={this.state.page}
                    total={this.state.total}
                    pageSize={this.state.size}
                    showPrevNextJumpers
                    locale={localeInfo}
                    onChange={this.onChange.bind(this)}
                />
            </div>
        );
    }
}

export default ProfessorsAvailability;
