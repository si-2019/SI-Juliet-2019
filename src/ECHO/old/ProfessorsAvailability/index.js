import ProfessorsAvailability from './ProfessorsAvailability';

export default ProfessorsAvailability;