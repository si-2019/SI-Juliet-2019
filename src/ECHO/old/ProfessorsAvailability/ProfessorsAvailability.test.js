import React from 'react';
import ProfessorsAvailability from './ProfessorsAvailability';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });


it("renders table", () => {
  const component = shallow(<ProfessorsAvailability />);
  expect(component).toMatchSnapshot();
});
it("renders search bar", () => {
  const wrapper = shallow(<ProfessorsAvailability />);
  expect(wrapper.find("#searchTeachingStaff").exists());
});
it("renders teaching staff table", () => {
  const wrapper = shallow(<ProfessorsAvailability />);
  expect(wrapper.find("#teachingStaffTable").exists());
});
it("renders pagination ", () => {
  const wrapper = shallow(<ProfessorsAvailability />);
  expect(wrapper.find("#pagination").exists()).toBe(true);
});