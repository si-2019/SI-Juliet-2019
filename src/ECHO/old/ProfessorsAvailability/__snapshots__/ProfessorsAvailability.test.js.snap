// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfessorsAvailability />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row search-teachers shadow-border"
        >
          <input
            onChange={[Function]}
            placeholder="Search.."
            type="text"
          />
        </div>,
        <div
          className="row"
        >
          <table
            className="table table-bordered table-hover professors-table shadow-border"
          >
            <thead>
              <tr>
                <th>
                  Name
                </th>
                <th>
                  Email
                </th>
                <th>
                  Title
                </th>
                <th />
              </tr>
            </thead>
            <tbody />
          </table>
        </div>,
        <Pagination
          className=""
          current={1}
          defaultCurrent={1}
          defaultPageSize={10}
          hideOnSinglePage={false}
          itemRender={[Function]}
          locale={
            Object {
              "items_per_page": "/ page",
              "jump_to": "Goto",
              "jump_to_confirm": "confirm",
              "next_3": "Next 3 Pages",
              "next_5": "Next 5 Pages",
              "next_page": "Next Page",
              "page": "",
              "prev_3": "Previous 3 Pages",
              "prev_5": "Previous 5 Pages",
              "prev_page": "Previous Page",
            }
          }
          onChange={[Function]}
          onShowSizeChange={[Function]}
          pageSize={10}
          prefixCls="rc-pagination"
          selectComponentClass={null}
          selectPrefixCls="rc-select"
          showLessItems={false}
          showPrevNextJumpers={true}
          showQuickJumper={false}
          showSizeChanger={false}
          showTitle={true}
          style={Object {}}
          total={0}
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onChange={[Function]}
            placeholder="Search.."
            type="text"
          />,
          "className": "row search-teachers shadow-border",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search..",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <table
            className="table table-bordered table-hover professors-table shadow-border"
          >
            <thead>
              <tr>
                <th>
                  Name
                </th>
                <th>
                  Email
                </th>
                <th>
                  Title
                </th>
                <th />
              </tr>
            </thead>
            <tbody />
          </table>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Email
                  </th>
                  <th>
                    Title
                  </th>
                  <th />
                </tr>
              </thead>,
              <tbody />,
            ],
            "className": "table table-bordered table-hover professors-table shadow-border",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Email
                  </th>
                  <th>
                    Title
                  </th>
                  <th />
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Name
                    </th>,
                    <th>
                      Email
                    </th>,
                    <th>
                      Title
                    </th>,
                    <th />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name",
                    },
                    "ref": null,
                    "rendered": "Name",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email",
                    },
                    "ref": null,
                    "rendered": "Email",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Title",
                    },
                    "ref": null,
                    "rendered": "Title",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "current": 1,
          "defaultCurrent": 1,
          "defaultPageSize": 10,
          "hideOnSinglePage": false,
          "itemRender": [Function],
          "locale": Object {
            "items_per_page": "/ page",
            "jump_to": "Goto",
            "jump_to_confirm": "confirm",
            "next_3": "Next 3 Pages",
            "next_5": "Next 5 Pages",
            "next_page": "Next Page",
            "page": "",
            "prev_3": "Previous 3 Pages",
            "prev_5": "Previous 5 Pages",
            "prev_page": "Previous Page",
          },
          "onChange": [Function],
          "onShowSizeChange": [Function],
          "pageSize": 10,
          "prefixCls": "rc-pagination",
          "selectComponentClass": null,
          "selectPrefixCls": "rc-select",
          "showLessItems": false,
          "showPrevNextJumpers": true,
          "showQuickJumper": false,
          "showSizeChanger": false,
          "showTitle": true,
          "style": Object {},
          "total": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row search-teachers shadow-border"
          >
            <input
              onChange={[Function]}
              placeholder="Search.."
              type="text"
            />
          </div>,
          <div
            className="row"
          >
            <table
              className="table table-bordered table-hover professors-table shadow-border"
            >
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Email
                  </th>
                  <th>
                    Title
                  </th>
                  <th />
                </tr>
              </thead>
              <tbody />
            </table>
          </div>,
          <Pagination
            className=""
            current={1}
            defaultCurrent={1}
            defaultPageSize={10}
            hideOnSinglePage={false}
            itemRender={[Function]}
            locale={
              Object {
                "items_per_page": "/ page",
                "jump_to": "Goto",
                "jump_to_confirm": "confirm",
                "next_3": "Next 3 Pages",
                "next_5": "Next 5 Pages",
                "next_page": "Next Page",
                "page": "",
                "prev_3": "Previous 3 Pages",
                "prev_5": "Previous 5 Pages",
                "prev_page": "Previous Page",
              }
            }
            onChange={[Function]}
            onShowSizeChange={[Function]}
            pageSize={10}
            prefixCls="rc-pagination"
            selectComponentClass={null}
            selectPrefixCls="rc-select"
            showLessItems={false}
            showPrevNextJumpers={true}
            showQuickJumper={false}
            showSizeChanger={false}
            showTitle={true}
            style={Object {}}
            total={0}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onChange={[Function]}
              placeholder="Search.."
              type="text"
            />,
            "className": "row search-teachers shadow-border",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Search..",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <table
              className="table table-bordered table-hover professors-table shadow-border"
            >
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Email
                  </th>
                  <th>
                    Title
                  </th>
                  <th />
                </tr>
              </thead>
              <tbody />
            </table>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <tr>
                    <th>
                      Name
                    </th>
                    <th>
                      Email
                    </th>
                    <th>
                      Title
                    </th>
                    <th />
                  </tr>
                </thead>,
                <tbody />,
              ],
              "className": "table table-bordered table-hover professors-table shadow-border",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <th>
                      Name
                    </th>
                    <th>
                      Email
                    </th>
                    <th>
                      Title
                    </th>
                    <th />
                  </tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <th>
                        Name
                      </th>,
                      <th>
                        Email
                      </th>,
                      <th>
                        Title
                      </th>,
                      <th />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Name",
                      },
                      "ref": null,
                      "rendered": "Name",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Email",
                      },
                      "ref": null,
                      "rendered": "Email",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Title",
                      },
                      "ref": null,
                      "rendered": "Title",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "th",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "tbody",
              },
            ],
            "type": "table",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "current": 1,
            "defaultCurrent": 1,
            "defaultPageSize": 10,
            "hideOnSinglePage": false,
            "itemRender": [Function],
            "locale": Object {
              "items_per_page": "/ page",
              "jump_to": "Goto",
              "jump_to_confirm": "confirm",
              "next_3": "Next 3 Pages",
              "next_5": "Next 5 Pages",
              "next_page": "Next Page",
              "page": "",
              "prev_3": "Previous 3 Pages",
              "prev_5": "Previous 5 Pages",
              "prev_page": "Previous Page",
            },
            "onChange": [Function],
            "onShowSizeChange": [Function],
            "pageSize": 10,
            "prefixCls": "rc-pagination",
            "selectComponentClass": null,
            "selectPrefixCls": "rc-select",
            "showLessItems": false,
            "showPrevNextJumpers": true,
            "showQuickJumper": false,
            "showSizeChanger": false,
            "showTitle": true,
            "style": Object {},
            "total": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
